lib/utils/display.js

```diff
@@ -3,6 +3,44 @@ const npmlog = require('npmlog')
 const log = require('./log-shim.js')
 const { explain } = require('./explain-eresolve.js')
 
+const originalCustomInspect = Symbol('npm.display.original.util.inspect.custom')
+
+// These are most assuredly not a mistake
+// https://eslint.org/docs/latest/rules/no-control-regex
+/* eslint-disable no-control-regex */
+// \x00 through \x1f, \x7f through \x9f, not including \x09 \x0a \x0b \x0d
+const hasC01 = /[\x00-\x08\x0c\x0e-\x1f\x7f-\x9f]/
+// Allows everything up to '[38;5;255m' in 8 bit notation
+const allowedSGR = /^\[[0-9;]{0,8}m/
+// '[38;5;255m'.length
+const sgrMaxLen = 10
+
+// Strips all ANSI C0 and C1 control characters (except for SGR up to 8 bit)
+function stripC01 (str) {
+  if (!hasC01.test(str)) {
+    return str
+  }
+  let result = ''
+  for (let i = 0; i < str.length; i++) {
+    const char = str[i]
+    const code = char.charCodeAt(0)
+    if (!hasC01.test(char)) {
+      // Most characters are in this set so continue early if we can
+      result = `${result}${char}`
+    } else if (code === 27 && allowedSGR.test(str.slice(i + 1, i + sgrMaxLen + 1))) {
+      // \x1b with allowed SGR
+      result = `${result}\x1b`
+    } else if (code <= 31) {
+      // escape all other C0 control characters besides \x7f
+      result = `${result}^${String.fromCharCode(code + 64)}`
+    } else {
+      // hasC01 ensures this is now a C1 control character or \x7f
+      result = `${result}^${String.fromCharCode(code - 64)}`
+    }
```

Comment:
💭 Would it be clearer to use the char representation: '\x1b', as opposed to 27?

```suggestion
    const char = str[i]
    if (!hasC01.test(char)) {
      // Most characters are in this set so continue early if we can
      result = `${result}${char}`
    } else if (char === '\x1b' && allowedSGR.test(str.slice(i + 1, i + sgrMaxLen + 1))) {
      // \x1b with allowed SGR
      result = `${result}\x1b`
    } else {
    

      // escape everything else
      const code = char.charCodeAt(0)
      if (code <= 31) {
        result = `${result}^${String.fromCharCode(code + 64)}`
      } else {
        // hasC01 ensures this is now a C1 control character or \x7f
        result = `${result}^${String.fromCharCode(code - 64)}`
      }
      
    }
```

Reply:
All of the other comparisons were numeric so I kept it consistent and added the comment for clarification of what character 27 is.